File: JULY-2025/19-07-2025/p1233.php
class Solution {

    /**
     * @param String[] $folder
     * @return String[]
     */
    function removeSubfolders($folder) {
        asort($folder);
        $folder = array_values($folder);
        $folderList=[$folder[0]];
        $i=1;
        $k=0;
        while($folder[$i]){
            if(strlen($folder[$i]) > $folderList[$k] && strpos($folder[$i],$folderList[$k].'/') === 0){
                $i++;
                continue;
            }
            $folderList[]=$folder[$i];
            $i++;
            $k++;
        }
        return $folderList;
        
    }
}

File: JULY-2025/19-07-2025/p1233.c
#include <stdlib.h>
#include <string.h>

int cmp(const void *a, const void *b) {
    return strcmp(*(const char **)a, *(const char **)b);
}

char **removeSubfolders(char **folder, int folderSize, int *returnSize) {
    qsort(folder, folderSize, sizeof(char *), cmp);
    char **res = (char **)malloc(folderSize * sizeof(char *));
    int cnt = 0;
    for (int i = 0; i < folderSize; ++i) {
        if (cnt == 0 || strncmp(folder[i], res[cnt-1], strlen(res[cnt-1])) != 0 ||
            folder[i][strlen(res[cnt-1])] != '/')
        {
            res[cnt++] = folder[i];
        }
    }
    *returnSize = cnt;
    return res;
}